cupyx/scipy/special/_beta.py

```diff
@@ -0,0 +1,1113 @@
+    }
+
+    y = Gamma(y);
+    a = Gamma(a);
+    b = Gamma(b);
+    if (y == 0.0) {
+        // goto overflow
+        return sign * CUDART_INF;
+    }
+
+    if (fabs(fabs(a) - fabs(y)) > fabs(fabs(b) - fabs(y))) {
+        y = b / y;
+        y *= a;
+    }
+    else {
+        y = a / y;
+        y *= b;
+    }
+
+    return (y);
+
+// overflow:
+//     sf_error("beta", SF_ERROR_OVERFLOW, NULL);
+//     return (sign * CUDART_INF);
+}
+"""
+
+
+lbeta_definition = """
+
+__device__ double lbeta(double, double);
+
+
+/*
+ * Special case for a negative integer argument
+ */
+
+__device__ static double lbeta_negint(int a, double b)
+{
+    double r;
+    if (b == (int)b && 1 - a - b > 0) {
+        r = lbeta(1 - a - b, b);
+        return r;
+    }
+    else {
+        // sf_error("lbeta", SF_ERROR_OVERFLOW, NULL);
+        return CUDART_INF;
+    }
+}
+
+// Natural log of |beta|
+
+__device__ double lbeta(double a, double b)
+{
+    double y;
+    int sign;
+
+    sign = 1;
+
+    if (a <= 0.0) {
+        if (a == floor(a)) {
+            if (a == (int)a) {
+                return lbeta_negint((int)a, b);
+            }
+            else {
+                // goto over;
+                return CUDART_INF;
+            }
+        }
+    }
+
+    if (b <= 0.0) {
+        if (b == floor(b)) {
+            if (b == (int)b) {
+                return lbeta_negint((int)b, a);
+            }
+            else {
+                // goto over;
+                return CUDART_INF;
+            }
+        }
+    }
+
+    if (fabs(a) < fabs(b)) {
+        y = a; a = b; b = y;
+    }
+
+    if (fabs(a) > ASYMP_FACTOR * fabs(b) && a > ASYMP_FACTOR) {
+        /* Avoid loss of precision in lgam(a + b) - lgam(a) */
+        y = lbeta_asymp(a, b, &sign);
+        return y;
+    }
+
+    y = a + b;
+    if (fabs(y) > MAXGAM || fabs(a) > MAXGAM || fabs(b) > MAXGAM) {
+        int sgngam;
+        y = lgam_sgn(y, &sgngam);
+        sign *= sgngam;     /* keep track of the sign */
+        y = lgam_sgn(b, &sgngam) - y;
+        sign *= sgngam;
+        y = lgam_sgn(a, &sgngam) + y;
+        sign *= sgngam;
+        return (y);
+    }
+
+    y = Gamma(y);
+    a = Gamma(a);
+    b = Gamma(b);
+    if (y == 0.0) {
+        // over:
+        // sf_error("lbeta", SF_ERROR_OVERFLOW, NULL);
+        return (sign * CUDART_INF);
+    }
+
+    if (fabs(fabs(a) - fabs(y)) > fabs(fabs(b) - fabs(y))) {
+        y = b / y;
+        y *= a;
+    }
+    else {
+        y = a / y;
+        y *= b;
+    }
+
+    if (y < 0) {
+    y = -y;
+    }
+
+    return (log(y));
+}
+"""
+
+
+beta = _core.create_ufunc(
+    "cupyx_scipy_beta",
+    ("ff->f", "dd->d"),
+    "out0 = out0_type(beta(in0, in1));",
+    preamble=(
+        beta_preamble +
+        gamma_definition +
+        polevl_definition +
+        p1evl_definition +
+        lgam_sgn_definition +
+        lbeta_symp_definition +
+        beta_definition
+    ),
+    doc="""Beta function.
+
+    Parameters
+    ----------
```

Comment:
Let's match docstring format to CuPy style :)

Reply:
I do see that I have an inconsistent mixture of both in the same docstring here where `.. seealso:: `` from the CuPy style was used when the rest of the docstring is in NumPy style, so I definitely need to make it self-consistent!

However, based on these comments, I thought NumPy style was being encouraged now? 
https://github.com/cupy/cupy/issues/5437#issuecomment-879148251
https://github.com/cupy/cupy/pull/6288#issuecomment-1010609903

For this PR it is not difficult to update to either style, please just confirm the preference so I can make any future PRs in the preferred style.

Reply:
Sorry for the late reply.

Oops, I completely overlooked this. Then it is ok to use NumPy style!

Thanks for the pointer and sorry for the confusion